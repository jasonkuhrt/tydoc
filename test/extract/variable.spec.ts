it('extracts docs', () => {
  expect(
    ctx.given(
      `
        export const a = 1
      `
    )
  ).toMatchSnapshot()
})

it('extracts the type name', () => {
  const result = ctx.givenVariables(
    `
      export const a = 1
      export const b:number = 1
      export let c = 1
      export const d = 'foo'
      export let e = 'foo'
    `
  )
  expect(result.terms.map(d => d.type.name)).toEqual([
    '1',
    'number',
    'number',
    '"foo"',
    'string',
  ])
})

describe('jsdoc', () => {
  it('is null when no jsDoc is present', () => {
    expect(
      ctx.given(
        `
          export const a = 1
        `
      ).terms[0].jsDoc
    ).toBeNull()
  })

  it('extracts doc from a variable statement', () => {
    expect(
      ctx.given(
        `
          /**
           * primary
           */
          export const a = 1
        `
      ).terms[0].jsDoc!.primary
    ).toMatchSnapshot()
  })

  it('splits multiple jsDoc blocks by primary and additional (closest to code is primary)', () => {
    expect(
      ctx.given(
        `
          /**
           * additional 2
           */
          /**
           * additional 1
           */
          /**
           * primary
           */
          export const a = 1
        `
      ).terms[0].jsDoc
    ).toMatchSnapshot()
  })

  it('whitespace and comments between multiple jsDoc blocks are ignored', () => {
    expect(
      ctx.given(
        `
          /**
           * additional 2
           */

           // boo 2

          /**
           * additional 1
           */

           // boo 1

          /**
           * primary
           */
          export const a = 1
        `
      ).terms[0].jsDoc
    ).toMatchSnapshot()
  })

  it.todo('keeps malformed jsdoc in raw form')
})
