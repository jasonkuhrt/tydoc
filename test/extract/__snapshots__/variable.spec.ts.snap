// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extracts docs 1`] = `
Object {
  "hybrids": Array [],
  "length": 1,
  "terms": Array [
    Object {
      "jsDoc": null,
      "kind": "variable",
      "languageLevel": "term",
      "name": "a",
      "sourceLocation": Object {
        "fileLine": 1,
        "filePath": "/test.ts",
      },
      "text": "a = 1",
      "textWithJSDoc": "a = 1",
      "type": Object {
        "name": "1",
      },
    },
  ],
  "types": Array [],
}
`;

exports[`jsdoc extracts doc from a variable statement 1`] = `
Object {
  "source": "primary",
}
`;

exports[`jsdoc splits multiple jsDoc blocks by primary and additional (closest to code is primary) 1`] = `
Object {
  "additional": Array [
    Object {
      "source": "additional 2",
    },
    Object {
      "source": "additional 1",
    },
  ],
  "primary": Object {
    "source": "primary",
  },
}
`;

exports[`jsdoc whitespace and comments between multiple jsDoc blocks are ignored 1`] = `
Object {
  "additional": Array [
    Object {
      "source": "additional 2",
    },
    Object {
      "source": "additional 1",
    },
  ],
  "primary": Object {
    "source": "primary",
  },
}
`;
